tasks/kubernetes.yaml

```diff
@@ -0,0 +1,55 @@
+---
+
+# This file is only for kubernetes specific actions related to deployment.
+# The main driver for separate files is the fact that, in kubernetes, a
+# TLS secret must be created for the broker. Logic shouldn't be added here
+# unless absolutely necessary.
+
+# TODO: Figure out how to use openssl from ansible
+#- name: Generate a Self Signed OpenSSL certificate
+#  openssl_certificate:
+#    path: /tmp/{{ broker_name }}-cert/cert.pem
+#    privatekey_path: /tmp/{{ broker_name }}-cert/key.pem
+#    subject: "/CN={{ broker_name }}.{{ broker_name }}.svc"
+#    provider: selfsigned
+- name: 'Create directory for OpenSSL cert'
+  file:
+    path: /tmp/{{ broker_name }}-cert
+    state: directory
+  when: cluster == 'kubernetes' and apb_action == 'provision'
```

Comment:
I think it's cleaner to encapsulate tasks that share a condition in a block or import task. Also prevents you from missing a conditional update somewhere if something changes.

Reply:
Yeah. I can definitely improve this file and the OpenShift file to encapsulate better.

Reply:
I believe I fixed it with the latest commit @fabianvf let me know what you think.